src/components/pages/production/ProductionStats.vue

```diff
@@ -0,0 +1,252 @@
+              taskTypeStatsMap[taskType.id].task_statuses
+            )"
+          ></div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+/*
+  This component displays the stats of a given production.
+  It shows the number of tasks done, the number of tasks in total, the total
+  duration spent on tasks and the total estimation of the tasks.
+  It also shows the repartition of the tasks by status as colored line.
+  When expanded, it shows the repartition of the tasks by task type.
+*/
+import { mapGetters } from 'vuex'
+
+import { formatListMixin } from '@/components/mixins/format'
+import TaskTypeName from '@/components/widgets/TaskTypeName'
+
+export default {
+  name: 'production-stats',
+
+  mixins: [formatListMixin],
+
+  components: {
+    TaskTypeName
+  },
+
+  props: {
+    stats: {
+      type: Object,
+      default: () => {}
+    }
+  },
+
+  data() {
+    return {
+      expanded: false
+    }
+  },
+
+  computed: {
+    ...mapGetters(['taskStatusMap', 'taskTypeMap']),
+
+    statusIds() {
+      return Object.keys(this.statsByStatus).sort((a, b) => {
+        return (
+          this.taskStatusMap.get(a).priority <
+          this.taskStatusMap.get(b).priority
+        )
+      })
+    },
+
+    taskTypes() {
+      const entityPriority = {
+        Asset: 1,
+        Shot: 2,
+        Sequence: 3,
+        Episode: 4,
+        Edit: 5,
+        Concept: 6
+      }
+      return Object.keys(this.taskTypeStatsMap)
+        .map(taskTypeId => {
+          return this.taskTypeMap.get(taskTypeId)
+        })
+        .sort((a, b) => {
+          if (a.for_entity !== b.for_entity) {
+            return entityPriority[a.for_entity] > entityPriority[b.for_entity]
+          } else {
+            return a.priority > b.priority
+          }
+        })
+    },
+
+    taskTypeStatsMap() {
+      const taskTypeStatsMap = {}
+      this.stats.task_types.forEach(taskTypeStat => {
+        if (!taskTypeStatsMap[taskTypeStat.task_type_id]) {
+          taskTypeStatsMap[taskTypeStat.task_type_id] = {
+            task_type_id: taskTypeStat.task_type_id,
+            amount: 0,
+            amount_done: 0,
+            total_duration: 0,
+            total_estimation: 0,
+            task_statuses: []
+          }
+        }
+        taskTypeStatsMap[taskTypeStat.task_type_id].amount +=
+          taskTypeStat.amount
+        taskTypeStatsMap[taskTypeStat.task_type_id].amount_done +=
+          taskTypeStat.amount_done
+        taskTypeStatsMap[taskTypeStat.task_type_id].total_duration +=
+          taskTypeStat.total_duration
+        taskTypeStatsMap[taskTypeStat.task_type_id].total_estimation +=
+          taskTypeStat.total_estimation
+        taskTypeStatsMap[taskTypeStat.task_type_id].task_statuses.push(
+          taskTypeStat
+        )
+      })
+      return taskTypeStatsMap
+    },
+
+    statsByStatus() {
+      const statsByStatus = {}
+      this.stats.task_types.forEach(taskTypeStat => {
```

Comment:
`this.stats.task_types` can be undefined if no stats are returned from the API for an empty production.